src/Exception/RequestException.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace Http\Client\Exception;
+
+use Psr\Http\Message\RequestInterface;
+
+/**
+ * Thrown when a request is available
+ *
+ * @author Márk Sági-Kazár <mark.sagikazar@gmail.com>
+ */
+class RequestException extends TransferException
+{
+    /**
+     * @var RequestInterface
+     */
+    protected $request;
```

Comment:
Well, we provide an extension of this class, where you might need to access this variable. But you are right, this should probably be private.